src/server/models/Reading.js

```diff
@@ -151,6 +156,34 @@ class Reading {
 	}
 
 	/**
+	 * Gets a number of compressed readings that approximate the given time range for the given groups.
+	 *
+	 * Compressed readings are in kilowatts.
+	 * @param groupIDs an array of ids for groups whose points are being compressed
+	 * @param {Moment?} fromTimestamp An optional start point for the time range.
+	 * @param {Moment?} toTimestamp An optional end point for the time range
+	 * @param numPoints The number of points to compress to. Defaults to 500
+	 * @param conn the connection to use. Defaults to the default database connection.
+	 * @return {Promise<object<int, array<{reading_rate: number, start_timestamp: Moment, end_timestamp: Moment}>>>}
+	 */
+	static async getCompressedGroupReadings(groupIDs, fromTimestamp = null, toTimestamp = null, numPoints = 500, conn = db) {
+		const allCompressedReadings = await conn.func('compressed_group_readings', [groupIDs, fromTimestamp || '-infinity', toTimestamp || 'infinity', numPoints]);
+		// Separate the result rows by meter_id and return a nested object.
+		const compressedReadingsByGroupID = mapToObject(groupIDs, () => []); // Returns { 1: [], 2: [], ... }
+		for (const row of allCompressedReadings) {
+			// HACK. Something in the SQL is causing these to be weird for some rows.
```

Comment:
This hack prevents an error from going to the client, but doesn't actually graph the group.

Reply:
It does in all cases I could produce. What groups do you have?

Reply:
Try Aldrich 2nd, Aldrich 3rd, Aldrich ground, Aldrich main, and Chapin 2nd with no child groups.

It's strange because a group containing Aldrich 1st, Aldrich 2nd, and Aldrich 3rd and no child groups graphs correctly.

Reply:
Even more bizarre, if I make a group that's a duplicate of the group that's failing, it works.